azure-ts-serverless-url-shortener-global/index.ts

```diff
@@ -0,0 +1,120 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as azure from "@pulumi/azure";
+import { Container } from "@azure/cosmos";
+import { getContainer } from "./cosmosclient";
+
+// Read a list of target locations from the config file:
+// Expecting a comma-separated list, e.g., "westus,eastus,westeurope"
+const config = new pulumi.Config();
+const locations = config.require("locations").split(',');
+// The first location is considered primary
+const primaryLocation = locations[0];
+
+let resourceGroup = new azure.core.ResourceGroup("urlshort-rg", {
+    location: primaryLocation,
+});
+
+// Cosmos DB with a single write region (primary location) and multiple read replicas
+let cosmosdb = new azure.cosmosdb.Account("url-cosmos", {
+    resourceGroupName: resourceGroup.name,
+    location: primaryLocation,
+    geoLocations: locations.map((location, priority) => ({
+        location: location,
+        failoverPriority: priority,
+    })),
```

Comment:
could be simplified as:  `locations.map((location, failoverPriority) => ({ location, failoverPriority })`